src/mon/OSDMonitor.cc

```diff
@@ -1430,20 +1433,20 @@ bool OSDMonitor::preprocess_failure(MOSDFailure *m)
   return false;
 
  didit:
-  m->put();
   return true;
 }
 
 class C_AckMarkedDown : public Context {
   OSDMonitor *osdmon;
-  MOSDMarkMeDown *m;
+  MonOpRequestRef op;
 public:
   C_AckMarkedDown(
     OSDMonitor *osdmon,
-    MOSDMarkMeDown *m)
-    : osdmon(osdmon), m(m) {}
+    MonOpRequestRef op)
+    : osdmon(osdmon), op(op) {}
 
   void finish(int) {
+    MOSDMarkMeDown *m = static_cast<MOSDMarkMeDown*>(op->get_req());
     osdmon->mon->send_reply(
       m,
```

Comment:
I wonder if we should funnel *all* replies through op->send_reply() (so that it can, say, mark itself) and rename the Monitor::send_reply to _send_reply or similar?